openedx_lti_tool_plugin/signals.py

```diff
@@ -24,3 +32,36 @@ def create_course_access_configuration(
     # Only create a CourseAccessConfiguration instance if LtiTool was created.
     if created:
         CourseAccessConfiguration.objects.get_or_create(lti_tool=instance)
+
+
+@receiver(course_grade_changed())
+def update_course_score(
+    sender: Any,  # pylint: disable=unused-argument
+    user: UserT,
+    course_grade: Any,
+    course_key: CourseKey,
+    **kwargs,
+):
+    """Update score for LTI graded resources with course ID as context key.
+
+    Args:
+        sender: Signal sender argument.
+        user: User instance.
+        course_grade: Course grade object.
+        course_key: Course opaque key.
+        **kwargs: Arbitrary keyword arguments.
+    """
+    # Ignore signal if plugin is disabled, is not a LTI user grade
+    # or the course grade has not been passed.
+    if (
+        not getattr(settings, 'OLTITP_ENABLE_LTI_TOOL', False)
+        or not getattr(user, 'openedx_lti_tool_plugin_lti_profile', None)
+        or not course_grade.passed
```

Comment:
Why only for passing grades?. @kuipumu

Reply:
@Jacatove There are two reasons:

1. We avoid querying for graded resources on each score update, this signal is called every time a user submits a problem, so it will send lots of updates all the time to all resources.
2. There are different statuses the AGS score can have when updating (https://www.imsglobal.org/spec/lti-ags/v2p0#gradingprogress, https://www.imsglobal.org/spec/lti-ags/v2p0#activityprogress) I want to avoid sending a score that might not be the final score of a course grade, I'm still reviewing this on more detail but it seems if we send other types of gradingprogress and activityprogress we could create confusion on the LTI platform.